src/main/java/book/store/dto/user/field/match/FieldMatch.java

```diff
@@ -0,0 +1,21 @@
+package book.store.dto.user.field.match;
+
+import jakarta.validation.Constraint;
+import jakarta.validation.Payload;
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Target({ElementType.TYPE})
+@Retention(RetentionPolicy.RUNTIME)
+@Constraint(validatedBy = FieldMatchValidator.class)
+@Documented
+public @interface FieldMatch {
```

Comment:
It's better to make it more generic - I can help you to change it if you like. Currently it is not exactly `FieldMatch`, but it should be called `PasswordsMatch`.
```suggestion
public @interface PasswordsMatch {
```